src/java/org/apache/cassandra/index/sai/plan/QueryController.java

```diff
@@ -527,14 +529,24 @@ private IteratorsAndIndexes getTopKRows(List<PrimaryKey> sourceKeys, RowFilter.E
     private List<CloseableIterator<ScoredPrimaryKey>> orderSstables(QueryViewBuilder.QueryView queryView, List<PrimaryKey> sourceKeys)
     {
         List<CloseableIterator<ScoredPrimaryKey>> results = new ArrayList<>();
-        for (var e : queryView.view.values())
+        long totalRows = queryView.view.keySet().stream().mapToLong(sstable -> sstable.getTotalRows()).sum();
+        queryView.view.forEach((sstable, expressions) ->
         {
+            // We expect the number of top results found in each sstable to be proportional to its number of rows
+            // we don't pad this number more because resuming a search if we guess too low is very very inexpensive.
+            //
+            // Note that this is only used for calls to `orderBy`, i.e., vanilla ANN with no other predicates;
+            // `orderResultsBy` does not know how to auto-resume.
```

Comment:
`orderResultsBy` does auto-resume, FWIW

Reply:
It's only the `BOUNDED_ANN` searches that do not auto-resume

Reply:
Thanks for clarifying, I'll revise.